pkg/handlers/handlers.go

```diff
@@ -38,7 +39,15 @@ type fileReader struct {
 	*iobuf.BufferedReadSeeker
 }
 
-var ErrEmptyReader = errors.New("reader is empty")
+var (
+	ErrEmptyReader = errors.New("reader is empty")
+
+	// ErrCriticalProcessing indicates a critical error that should halt processing.
```

Comment:
"Critical" is one of those words that unfortunately doesn't really convey any information to someone who doesn't already know the domain. Should these errors halt processing of the source? The file? Something else?

Reply:
Yea, I agree. I'll update it. 👍